server/src/main/java/org/opensearch/index/mapper/DocumentParser.java

```diff
@@ -592,9 +600,19 @@ private static void parseArray(ParseContext context, ObjectMapper parentMapper,
                 parentMapper = parentMapperTuple.v2();
                 ObjectMapper.Dynamic dynamic = dynamicOrDefault(parentMapper, context);
                 if (dynamic == ObjectMapper.Dynamic.STRICT) {
-                    throw new StrictDynamicMappingException(parentMapper.fullPath(), arrayFieldName);
-                } else if (dynamic == ObjectMapper.Dynamic.TRUE) {
-                    Mapper.Builder builder = context.root().findTemplateBuilder(context, arrayFieldName, XContentFieldType.OBJECT);
+                    throw new StrictDynamicMappingException(
+                        dynamic.name().toLowerCase(Locale.ROOT),
+                        parentMapper.fullPath(),
+                        arrayFieldName
+                    );
+                } else if (dynamic == ObjectMapper.Dynamic.TRUE || dynamic == ObjectMapper.Dynamic.STRICT_ALLOW_TEMPLATES) {
```

Comment:
Style nitpick, but would this "if-else-if-with-an-or-condition-else` structure be more easily understandable as a switch statement? Same applies for the other place this condition was added. I don't feel strongly so I'll leave it up to your preference.

Reply:
We use both `if-else` and `switch-case` in different methods here, it's better that we make them consistent,  I've changed `if-else` to `switch-case` in two places, thank you!